docs/SetupAzureDatabase-1.md

```diff
@@ -0,0 +1,58 @@
+# Set up a database
+
+This guide explains how to set up a database for Scalar DL in Azure and how to configure helm charts for Scalar DL deployment.
+
+## Cosmos DB
+
+In this section, you will set up a Cosmos DB for Scalar DL.
+
+### Requirements
+
+* You must create a Cosmos DB account with `Core(SQL)` API.
+* You must change Cosmos DB `Default consistency` to `Strong`.
+* You must use Capacity mode as `Provisioned throughput`.
+
+### Steps
+
+* Create an Azure Cosmos DB account on the basis of [Azure official guide](https://docs.microsoft.com/en-us/azure/cosmos-db/create-cosmosdb-resources-portal#create-an-azure-cosmos-db-account) with the above requirements.
+
+### Configure Scalar DL 
+
+To apply the Scalar DL schema to Cosmos DB, update the following configuration in [schema-loading-custom-values](https://github.com/scalar-labs/scalar-kubernetes/blob/master/conf/schema-loading-custom-values.yaml)
+
+```yaml
+database: cosmos
+contactPoints: <Cosmos DB account endpoint>
+password: <Cosmos DB account primary master key>
```

Comment:
If users write the password directly in `values.yaml`, they will need to treat this entire file as a credential, and configuration and credentials cannot be separated. This may not be a problem in a development environment, but it should be avoided in a production environment. 

Users can use a Secret object created separately instead of writing the password directly in `values.yaml`. It would be better to include it as a Tip.

For example, I would suggest the following sentence. (English should be modified as necessary.)

### Tip: Separate credentials from configuration

If you want to separate DB username and password from the configuration file, you can also use the existing Secret object.

First, you create a Secret object. The key to store the account name should be `db-username` and the key to store the password should be `db-password`.

```console
$ kubectl create secret generic scalardl --from-literal db-username=<Cosmos DB account name> --from-literal db-password=<Cosmos DB account primary master key>
```

Next, in each configuration file, put the name of the Secret object in `existingSecret`.

**schema-loading**

```yaml
schemaLoading:
  existingSecret: scalardl
```

**scalardl**

```yaml
ledger:
  existingSecret: scalardl
```

@feeblefakie WDYT?

Reply:
@superbrothers Thank you! That sounds good.
@mathewseby I think it should be added to the recommendation section, so please do.

Reply:
@feeblefakie san,
>  I think it should be added to the recommendation section, so please do.

Yes, I will do it.

> But, I don't think it's specific to Cosmos DB, so we can create a general doc and refer to it from here?

Create a separate document only for this section?

Reply:
@mathewseby Please try and see how it goes. Reorganize the docs before actually starting it.

Reply:
@superbrothers san,
Thank you for the suggestion.
When I tried schema loading deployment it failed with following error
```
Exception in thread "main" java.lang.IllegalArgumentException: Failed to add contact point: cassandra
	at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:922)
	at com.datastax.driver.core.Cluster$Builder.addContactPoints(Cluster.java:942)
	at qbits.alia.cluster_options$fn__1049.invokeStatic(cluster_options.clj:41)
	at qbits.alia.cluster_options$fn__1049.invoke(cluster_options.clj:39)
	at clojure.lang.MultiFn.invoke(MultiFn.java:239)
	at qbits.alia.cluster_options$set_cluster_options_BANG_$fn__1170.invoke(cluster_options.clj:256)
	at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49)
	at clojure.core.protocols$fn__8140.invokeStatic(protocols.clj:75)
	at clojure.core.protocols$fn__8140.invoke(protocols.clj:75)
	at clojure.core.protocols$fn__8088$G__8083__8101.invoke(protocols.clj:13)
	at clojure.core$reduce.invokeStatic(core.clj:6828)
	at qbits.alia.cluster_options$set_cluster_options_BANG_.invokeStatic(cluster_options.clj:252)
	at qbits.alia$cluster.invokeStatic(alia.clj:259)
	at scalar_schema.cassandra$get_cluster.invokeStatic(cassandra.clj:14)
	at scalar_schema.cassandra$make_cassandra_operator.invokeStatic(cassandra.clj:78)
	at scalar_schema.operations$make_operator.invokeStatic(operations.clj:11)
	at scalar_schema.operations$create_tables.invokeStatic(operations.clj:17)
	at scalar_schema.core$_main.invokeStatic(core.clj:32)
	at scalar_schema.core$_main.doInvoke(core.clj:29)
	at clojure.lang.RestFn.applyTo(RestFn.java:137)
	at scalar_schema.core.main(Unknown Source)
Caused by: java.net.UnknownHostException: cassandra: Name or service not known
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
	at java.net.InetAddress.getAllByName(InetAddress.java:1193)
	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
	at com.datastax.driver.core.Cluster$Builder.addContactPoint(Cluster.java:919)
	... 20 more 
```
Is there any additional configuration needed for specifying the database?

Reply:
The command I presented was wrong. The correct one is below :bow:

```
$ kubectl create secret generic scalardl --from-literal username=<Cosmos DB account name> --from-literal password=<Cosmos DB account primary master key>
```

Reply:
@superbrothers san,
> The command I presented was wrong. The correct one is below

Thank you.
But, when I tried with that command one I am getting `CreateContainerConfigError`  
```
$ kubectl get pods
NAME                               READY   STATUS                       RESTARTS   AGE
load-schema-schema-loading-2lqfh   0/1     CreateContainerConfigError   0          37s
```
I think first one is the correct command but it is taking Cassandra database thats why we are getting [this](https://github.com/scalar-labs/scalar-kubernetes/pull/133#discussion_r669263721) error.

Following is the describe pod output when I used the [first command](https://github.com/scalar-labs/scalar-kubernetes/pull/133#discussion_r654733923)
```
$ kubectl describe pods load-schema-schema-loading-lzt27
Name:         load-schema-schema-loading-lzt27
Namespace:    default
Priority:     0
Node:         aks-scalardlpool-40363514-vmss000000/10.240.0.7
Start Time:   Wed, 14 Jul 2021 20:31:15 +0530
Labels:       app.kubernetes.io/instance=load-schema
              app.kubernetes.io/name=schema-loading
              controller-uid=156d0a96-a901-4e27-a9a7-9b916d38bb67
              job-name=load-schema-schema-loading
Annotations:  <none>
Status:       Failed
IP:           10.244.5.3
IPs:
  IP:           10.244.5.3
Controlled By:  Job/load-schema-schema-loading
Containers:
  schema-import:
    Container ID:  containerd://d5645fe679d2b7397de78cb43208bdeacc94b0dcf32ca55d8828bd084c4c6d8e
    Image:         ghcr.io/scalar-labs/scalardl-schema-loader:1.3.0
    Image ID:      ghcr.io/scalar-labs/scalardl-schema-loader@sha256:b5e9b0df8e85b2617443342a937a4df51d842bc3ce4ef35c4ac5f22e34889b6b
    Port:          <none>
    Host Port:     <none>
    Args:
      -P
      9042
      -u
      $(DB_USERNAME)
      -p
      $(DB_PASSWORD)
      --cassandra
      -h
      cassandra
      -n
      NetworkTopologyStrategy
      -R
      3
    State:          Terminated
      Reason:       Error
      Exit Code:    1
      Started:      Wed, 14 Jul 2021 20:31:26 +0530
      Finished:     Wed, 14 Jul 2021 20:31:29 +0530
    Ready:          False
    Restart Count:  0
    Environment:
      DB_USERNAME:  <set to the key 'db-username' in secret 'scalardl'>  Optional: false
      DB_PASSWORD:  <set to the key 'db-password' in secret 'scalardl'>  Optional: false
    Mounts:
      /var/run/secrets/kubernetes.io/serviceaccount from default-token-jljcz (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  default-token-jljcz:
    Type:        Secret (a volume populated by a Secret)
    SecretName:  default-token-jljcz
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  70s   default-scheduler  Successfully assigned default/load-schema-schema-loading-lzt27 to aks-scalardlpool-40363514-vmss000000
  Normal  Pulling    69s   kubelet            Pulling image "ghcr.io/scalar-labs/scalardl-schema-loader:1.3.0"
  Normal  Pulled     62s   kubelet            Successfully pulled image "ghcr.io/scalar-labs/scalardl-schema-loader:1.3.0" in 7.067621698s
  Normal  Created    59s   kubelet            Created container schema-import
  Normal  Started    59s   kubelet            Started container schema-import
```

Reply:
@superbrothers 
I have updated the manual guide with above specified changes (Separate credentials from configuration).
Could you please check when you get a chance? ae0abbffa032adf0f4d8e3edbd63506850722d8f